checkstyle-tester/diff.groovy

```diff
@@ -628,36 +617,40 @@ def getProjectsStatistic(diffDir) {
     return projectsStatistic
 }
 
-def runMavenExecution(srcDir, excludes, checkstyleConfig,
-                      checkstyleVersion, extraMvnRegressionOptions) {
-    println "Running 'mvn clean' on $srcDir ..."
-    def mvnClean = "mvn -e --no-transfer-progress --batch-mode clean"
-    executeCmd(mvnClean)
-    println "Running Checkstyle on $srcDir ... with excludes {$excludes}"
-    def mvnSite = "mvn -e --no-transfer-progress --batch-mode site " +
-        "-Dcheckstyle.config.location=$checkstyleConfig -Dcheckstyle.excludes=$excludes"
-    if (checkstyleVersion) {
-        mvnSite = mvnSite + " -Dcheckstyle.version=$checkstyleVersion"
+def runCliExecution(allJar, srcDir, excludes, checkstyleConfig, saveDir, continueOnError,
+                      extraRegressionOptions) {
+    def saveLoc = new File(saveDir)
+    if (!saveLoc.exists()) {
+        saveLoc.mkdirs()
     }
-    if (extraMvnRegressionOptions) {
-        mvnSite = mvnSite + " "
 
-        if (!extraMvnRegressionOptions.startsWith("-")) {
-            mvnSite = mvnSite + "-"
+    println "Running Checkstyle CLI on $srcDir ... with excludes {$excludes}"
+    def cliCommand = "java -jar $allJar -c $checkstyleConfig $srcDir -f xml -o " +
+        getOsSpecificPath("$saveDir", "checkstyle-result.xml") +
+        " -e " + getOsSpecificPath("$srcDir", ".git") +
+        " -e " + getOsSpecificPath("$srcDir", ".hg")
+
+    if (excludes.length() > 0) {
+        for (String singleExclude : excludes.split(',')) {
+            if (isWindows()) {
+                singleExclude = singleExclude.replace("/", "\\")
+            }
+            cliCommand = cliCommand + " -x " + singleExclude
         }
-        mvnSite = mvnSite + extraMvnRegressionOptions
     }
-    println(mvnSite)
-    executeCmd(mvnSite)
-    println "Running Checkstyle on $srcDir - finished"
-}
 
-def postProcessCheckstyleReport(targetDir, repoName, repoPath) {
-    new AntBuilder().replace(
-        file: getOsSpecificPath("$targetDir", "checkstyle-result.xml"),
-        token: new File(getOsSpecificPath("src", "main", "java", "$repoName")).absolutePath,
-        value: getOsSpecificPath("$repoPath")
-    )
+    if (extraRegressionOptions) {
+        cliCommand = cliCommand + " "
+
+        if (!extraRegressionOptions.startsWith("-")) {
+            cliCommand = cliCommand + "-"
+        }
+        cliCommand = cliCommand + extraRegressionOptions
+    }
+
+    println(cliCommand)
```

Comment:
Let's put some words before command

Reply:
We do have some words at https://github.com/checkstyle/contribution/blob/master/checkstyle-tester/diff.groovy#L636 .

Reply:
It usually bad to print something in out put without context what is it .

Reply:
We do double print now, second is few lines below.
```
def executeCliCmd(cmd, continueOnError) {
    println "Running command: ${cmd}"
``
So let's remove `println(cliCommand)`

Reply:
Please remove this line  is very clear double printing of command.

Reply:
Done.